squad/core/models.py

```diff
@@ -232,12 +232,22 @@ def test_suites_by_environment(self):
         result = OrderedDict()
         envlist = set([t.environment for t in test_runs])
         for env in sorted(envlist, key=lambda env: env.slug):
-            result[env] = set()
+            result[env] = dict()
         for tr in test_runs:
             for t in tr.tests.all():
-                result[tr.environment].add(t.suite)
+                if t.suite in result[tr.environment].keys():
+                    if t.status in result[tr.environment][t.suite].keys():
+                        result[tr.environment][t.suite][t.status] += 1
+                    else:
+                        result[tr.environment][t.suite].update({t.status: 1})
+                else:
+                    result[tr.environment].update({t.suite: {t.status: 1}})
         for env in result.keys():
-            result[env] = sorted(result[env], key=lambda suite: suite.slug)
+            # there should only be one key in the most nested dict
+            result[env] = sorted(
+                result[env].items(),
+                key=lambda suite_dict: next(iter(suite_dict)).slug)
+
```

Comment:
why not just `suite_dict[0].slug` like you did in the tests?

Reply:
I had dict of dicts here and that didn't work (sorting was commented)

Reply:
Fixed